src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/GenericWhitespaceCheck.java

```diff
@@ -219,7 +219,7 @@ private void processSingleGeneric(DetailAST ast, String line, int after) {
         //    Collections.<Object>emptySet();
         //                        ^
         //                        +--- whitespace not allowed
-        if (isGenericBeforeMethod(ast)) {
+        if (isGenericBeforeMethod(ast) || isGenericBeforeCtor(ast)) {
```

Comment:
Please remove upper comment completely.
It now becomes a bit misleading.

Please update doc https://checkstyle.org/config_whitespace.html#GenericWhitespace at description to make it
```
- should not be preceded with whitespace when it is precede method name or constructor
- should not be preceded with whitespace when following type name.
```

And 
```
should be followed with whitespace in almost all cases, except diamond operators and when preceding method name or contructor.
```

Please add example of correct ctor to `Examples with correct spacing:` under example for method

Reply:
Done